applications/FluidDynamicsApplication/custom_processes/weighted_divergence_calculation_process.h

```diff
@@ -0,0 +1,338 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Tosi
+
+#ifndef KRATOS_WEIGHTED_DIVERGENCE_CALCULATION_PROCESS_H
+#define KRATOS_WEIGHTED_DIVERGENCE_CALCULATION_PROCESS_H
+
+
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "includes/element.h"
+#include "includes/kratos_flags.h"
+#include "processes/process.h"
+#include "geometries/geometry.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/math_utils.h"
+#include "includes/kratos_parameters.h"
+#include "utilities/variable_utils.h"
+
+#include <string>
+#include <iostream>
+#include <sstream>
+
+#include <boost/functional/hash.hpp> //TODO: remove this dependence when Kratos has en internal one
+#include <unordered_map> //TODO: remove this dependence when Kratos has en internal one
+#include <utility>
+
+namespace Kratos
+{
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// This process computes the element average in time of the divergence and of the seminorm of the velocity field.
+/// We define the seminorm of the velocity field as: \left \| \nabla u_{h} \right \|_{L^2(K)}^2 ,
+/// and the divergence as \left \| \nabla \cdot u_{h} \right \|_{L^2(K)}^2 ,
+/// where u is the velocity field and K an element of the domain \Omega.
+/// The time average does not consider the transient 20% first part of the simulation.
+/// The process requires a model part as input.
+
+/** Detail class definition.
+ */
+
+class WeightedDivergenceCalculationProcess: public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of Process
+    KRATOS_CLASS_POINTER_DEFINITION(WeightedDivergenceCalculationProcess);
+
+    typedef ModelPart::ElementType ElementType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor for WeightedDivergenceCalculationProcess Process
+    WeightedDivergenceCalculationProcess(ModelPart& rModelPart
+                    ):
+        Process(),
+        mrModelPart(rModelPart)
+    {
+    }
+
+    /// Destructor.
+    ~WeightedDivergenceCalculationProcess() override {}
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// This operator is provided to call the process as a function and simply calls the Execute method.
+    void operator()()
+    {
+        Execute();
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Execution of the process
+    void Execute() override
```

Comment:
This method is extremely long. Is it possible to reduce it by using auxiliary private methods? (the good practice says that functions shouldn't have more than 100 lines, but of course this is not always possible).

Reply:
I am now around 90 95 lines. Tell me if you agree with the current splitting in smaller function, or you have some different suggestion. Thanks!

Reply:
Well, this is not a sort of threshold. If you feel that there are some tasks that can be an auxiliary method, it is a good practice to do it for the sake of code readability.

Reply:
Ok thanks!